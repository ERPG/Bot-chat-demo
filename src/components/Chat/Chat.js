import React from 'react';
import InputWrapper from './InputWrapper/InputWrapper';
import MessagesWrapper from './MessagesWrapper/MessagesWrapper';
import Header from './Header/Header';

/** For this component I have decided giving the layout of the app and with
 * the idea to make it as customizable as possible, to divide the component in three parts. */

export default function Chat() {
  return (
    <div data-landbot="chat-container">
      <Header />
      <MessagesWrapper />
      <InputWrapper />
    </div>
  );
}