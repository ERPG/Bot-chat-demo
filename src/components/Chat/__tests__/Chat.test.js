import React from 'react';
import { render } from '@testing-library/react';
import { customRender } from '../../../../__tests__/__utils__/testUtils';
import Chat from '../Chat';

describe('Chat component', () => {
  const chatContainerSelector = 'chat-container';

  beforeEach(() => {
    window.HTMLElement.prototype.scrollTo = jest.fn();
  });

  it('Should render component without errors', () => {
    const { container } = render(<Chat />);
    expect(container).toMatchSnapshot();
  });

  it('Should have a container', () => {
    const { getByDataLb } = customRender(<Chat />);
    const element = getByDataLb(chatContainerSelector)
    expect(element).toBeDefined();
  });
});