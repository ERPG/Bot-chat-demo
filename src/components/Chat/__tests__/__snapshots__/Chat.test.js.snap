// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component Should render component without errors 1`] = `
<div>
  <div
    data-landbot="chat-container"
  >
    <div>
      <div
        class="landbot-header"
        data-landbot="header-container"
      >
        <h1
          class="subtitle"
          data-landbot="header-subtitle"
        >
          Landbot
        </h1>
      </div>
    </div>
    <div
      class="landbot-messages-container"
      data-landbot="message-container"
      id="landbot-messages-container"
    />
    <div
      class="landbot-input-container"
      data-landbot="input-container"
    >
      <div
        class="field"
      >
        <div
          class="control"
        >
          <input
            class="landbot-input"
            data-landbot="chat-input"
            placeholder="Type here..."
            type="text"
            value=""
          />
          <button
            class="button landbot-input-send"
            data-landbot="send-button"
            disabled=""
          >
            <span
              class="icon is-large"
            >
              <i
                class="fas fa-paper-plane fa-lg"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
